src/sqlfluff/dialects/dialect_hive.py

```diff
@@ -1017,3 +1017,55 @@ class SortByClauseSegment(ansi.OrderByClauseSegment):
         ),
         Dedent,
     )
+
+
+class AlterTableStatementSegment(ansi.AlterTableStatementSegment):
+    """An `ALTER TABLE` statement.
+
+    Full Apache Hive `CREATE ALTER` reference here:
+    https://cwiki.apache.org/confluence/display/Hive/LanguageManual+DDL#LanguageManualDDL-AlterTable
+    """
+
+    match_grammar: Matchable = Sequence(
+        "ALTER",
+        "TABLE",
+        Ref("TableReferenceSegment"),
+        Delimited(
+            OneOf(
+                # Table options
+                Sequence(
+                    Ref("ParameterNameSegment"),
+                    Ref("EqualsSegment", optional=True),
+                    OneOf(Ref("LiteralGrammar"), Ref("NakedIdentifierSegment")),
+                ),
+                # Add things
+                Sequence(
+                    OneOf("ADD", "MODIFY"),
+                    Ref.keyword("COLUMN", optional=True),
+                    Ref("ColumnDefinitionSegment"),
+                    OneOf(
+                        Sequence(
+                            OneOf("FIRST", "AFTER"), Ref("ColumnReferenceSegment")
+                        ),
+                        # Bracketed Version of the same
+                        Ref("BracketedColumnReferenceListGrammar"),
+                        optional=True,
+                    ),
+                ),
+                # Rename
+                Sequence(
+                    "RENAME",
+                    OneOf("AS", "TO", optional=True),
+                    Ref("TableReferenceSegment"),
+                ),
```

Comment:
This part seems all the same as ANSI. I wonder if we could redefine that as `AlterTableOptionsGrammar` or something in ANSI and then just extend in Hive to add the "exchange" part, similar to: https://github.com/sqlfluff/sqlfluff/blob/72fed8048248426a7b6e95880cec87641249db06/src/sqlfluff/dialects/dialect_hive.py#L157-L166

Reply:
Hi @greg-finley thank you for the suggestion, should I do the same replacement in other dialects where the `ansi` `AlterTableStatementSegment` class is overwritten?

Reply:
If you find similar ways that they are copying the same code, sure! It can be a separate PR.